docs/src/appInsights.ts

```diff
@@ -0,0 +1,32 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+// eslint-disable-next-line import/no-unresolved
+import siteConfig from "@generated/docusaurus.config";
+import { ReactPlugin } from "@microsoft/applicationinsights-react-js";
+import { ApplicationInsights } from "@microsoft/applicationinsights-web";
+
+const reactPlugin = new ReactPlugin();
+let appInsights: ApplicationInsights | undefined;
+
+if (siteConfig?.customFields?.INSTRUMENTATION_KEY === undefined) {
+	console.warn("Instrumentation Key is missing. App Insights will not be initialized.");
+}
+
+// Only initialize Application Insights if not in local development.
+// Remove the condition if you want to run Application Insights locally.
```

Comment:
Can you create a condition that handles running the AppInsights on local environment (maybe perhaps create a flag in the config)? Manually removing the condition to test locally seems to be error prone.